frontend/src/components/CalcSpectrum.tsx

```diff
@@ -185,8 +248,73 @@ export const CalcSpectrum: React.FC = () => {
     } else {
       setValue("simulate_slit", 5);
     }
-  }, [databaseWatch, modeWatch]);
-
+    if (isDirty === true) {
+      setDownloadButton(true);
+    }
+    if (tgasWatch !== 300) {
+      setDownloadButton(true);
+    }
+    if (pressureWatch !== 1.01325) {
+      setDownloadButton(true);
+    }
+    if (pathLengthWatch !== 1) {
+      setDownloadButton(true);
+    }
+    if (tvibWathch === 300 || tvibWathch === undefined) {
+      setDownloadButton(true);
+    }
+    if (trotWathch === 300 || trotWathch === undefined) {
+      setDownloadButton(true);
+    }
+    if (slitWatch === 5 && slitWatch === undefined) {
+      setDownloadButton(true);
+    }
+    if (minWaveRangeWatch !== 1900) {
+      setDownloadButton(true);
+    }
+    if (maxWaveRangeWatch !== 2300) {
+      setDownloadButton(true);
+    }
```

Comment:
I think there should be a way to do this without having to hard-code every default values here.

Reply:
I'll take a closer look at this by tomorrow. Is there a deadline approaching that you need to merge this by @arunavabasu-03? Otherwise, I'd really like to see if we can refactor this frontend a little bit to make it possible to have a simple way to disable/enable the download button.

Reply:
@suzil I try to do that but it is not perfectly working

Reply:
and in the field array it is more tricky so i done that way

Reply:
nope @suzil, take your time

Reply:
Actually, there is a major conference next week starting Wednesday and I'll have a talk to present Radis.

I'll mention Radis App so it's good if we can merge.

And @arunavabasu-03 you may refactor a better implementation afterwards.

Reply:
Actually, I have an idea for your Gray button.

Just turn it gray whenever you press Calculate.
Then visible when the spectrum is calculated and shown. 
Then gray again when Calculate is pressed again.

Should be easier to implement, and good user experience